tests/FunctionalTestCase.php

```diff
@@ -114,6 +114,26 @@ protected function getServerStorageEngine(ReadPreference $readPreference = null)
         throw new UnexpectedValueException('Could not determine server storage engine');
     }
 
+    protected function skipIfCausalConsistencyIsNotSupported()
+    {
+        if ($this->getPrimaryServer()->getType() === Server::TYPE_STANDALONE) {
+            $this->markTestSkipped('Causal Consistency is not supported on standalone servers');
+        }
+
+        if ($this->getPrimaryServer()->getType() === Server::TYPE_MONGOS) {
+            if (version_compare($this->getServerVersion(), '3.6.0', '<')) {
+                $this->markTestSkipped('Causal Consistency is only supported on MongoDB 3.6 or higher');
+            }
+        } else {
```

Comment:
Rather than nest conditionals, I propose we use an early return after the version compare conditional.

Alternatively, you can switch on the server type and handle mongos and a replica set explicitly. Then, `default:` can always `markTestSkipped()`.